test/box-luatest/gh_6150_memtx_tx_memory_monitoring_test.lua

```diff
@@ -0,0 +1,200 @@
+    }
+    tx_step(g.server, 'tx1', "s:replace{1, 1}", diff)
+    diff = {
+        ["txn"] = {
+            ["statements"] = {
+                ["avg"] = 68,
+                ["total"] = 136,
+            },
+            ["system data"] = {
+                ["total"] = 147,
+                ["avg"] = 74
+            }
+        },
+        ["memtx mvcc"] = {
+            ["tuple stories"] = {
+                ["directly used by txn"] = {
+                    ["stories total"] = 152,
+                    ["stories count"] = 1,
+                    ["retained tuples count"] = 1,
+                    ["retained tuples total"] = 9,
+                }
+            }
+        }
+    }
+    tx_step(g.server, 'tx2', "s:replace{1, 2}", diff)
+    diff = {
+        ["txn"] = {
+            ["statements"] = {
+                ["total"] = 136,
+                ["avg"] = 68,
+                ["max"] = 136,
+            },
+            ["system data"] = {
+                ["total"] = 147,
+                ["avg"] = 73,
+                ["max"] = 147,
+            }
+        },
+        ["memtx mvcc"] = {
+            ["tuple stories"] = {
+                ["directly used by txn"] = {
+                    ["stories total"] = 152,
+                    ["stories count"] = 1,
+                }
+            }
+        }
+    }
+    tx_step(g.server, 'tx2', "s:replace{2, 2}", diff)
+    tx_gc(g.server, 100, nil)
+    local err = g.server:eval('return tx2:commit()')
+    assert(not err[1])
+    diff = {
+        ["txn"] = {
+            ["statements"] = {
+                ["total"] = -136 * 2,
+                ["avg"] = -68,
+                ["max"] = -136,
+            },
+            ["system data"] = {
+                ["total"] = -147 * 2,
+                ["avg"] = -73,
+                ["max"] = -147,
+            }
+        },
+        ["memtx mvcc"] = {
+            ["tuple stories"] = {
+                ["directly used by txn"] = {
+                    ["stories total"] = -152,
+                    ["stories count"] = -1,
+                }
+            }
+        }
+    }
+    tx_gc(g.server, 10, diff)
+    print(string.rep('------------------\n', 5))
+    err = g.server:eval('return tx1:commit()')
+    assert(not err[1])
+    diff = {
+        ["txn"] = {
+            ["statements"] = {
+                ["total"] = -136,
+                ["avg"] = -136,
+                ["max"] = -136,
+            },
+            ["system data"] = {
+                ["total"] = -147,
+                ["avg"] = -147,
+                ["max"] = -147,
+            }
+        },
+        ["memtx mvcc"] = {
+            ["tuple stories"] = {
+                ["directly used by txn"] = {
+                    ["stories total"] = -152 * 2,
+                    ["stories count"] = -2,
+                    ["retained tuples total"] = -9,
+                    ["retained tuples count"] = -1,
+                }
+            }
+        }
+    }
+    tx_gc(g.server, 100, diff)
+end
```

Comment:
Why does your test cover only a small subset of statistics?

Reply:
What about `txn.user`?

Reply:
It is used only by external modules.

Reply:
Added new test